platform/windows/display_server_windows.cpp

```diff
@@ -1190,7 +1195,7 @@ void DisplayServerWindows::_update_window_style(WindowID p_window, bool p_repain
 	SetWindowLongPtr(wd.hWnd, GWL_EXSTYLE, style_ex);
 
 	if (icon.is_valid()) {
-		set_icon(icon);
+		window_set_icon(icon, p_window);
```

Comment:
This will reset every window to the last set icon, so `icon` should be moved to `WindowData`.